File: Design Matters Part-Three: Forever Single/Forever Single in C#/Adapter/PGP1TOPGP2Adapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Adapter
{
    class PGP1TOPGP2Adapter : PGP2Interface
    {

        PGPClass pgpObject;
        string fname;
        string lname;

        public PGP1TOPGP2Adapter(PGPClass a)
        {
            pgpObject = a;
            fname = pgpObject.getName().Split(' ')[0];
            lname = pgpObject.getName().Split(' ')[1];

        }
        
        public string getFirstName()
        {
            return fname;
        }

        public string getLastName()
        {
            return lname;
        }

        public void setFirstName(string f)
        {
            fname = f;
        }

        public void setLastName(string l)
        {
            lname = l;
        }
    }
}

File: Design Matters Part-Two: Adapter Party/Adapter Party in Java/PGP1TOPGP2Adapter.java
public class PGP1TOPGP2Adapter Implement PGP2Interface {

	public PGPClass pgpObject;
	public String fname;
	public String lname;
	
	public PGP1TOPGP2Adapter(PGPClass pgpObject)
	{
		this.pgpObject = pgpObject;
		fname = pgpObject.getName().Split(' ')[0];
            	lname = pgpObject.getName().Split(' ')[1];
	}
	
	public string getFirstName()
        {
            return this.fname;
        }

        public string getLastName()
        {
            this.return lname;
        }

        public void setFirstName(string fname)
        {
            this.fname = fname;
        }

        public void setLastName(string lname)
        {
            this.lname = lname;
        }
	
}